oml/lightning/entrypoints/train.py

```diff
@@ -128,33 +51,65 @@ def pl_train(cfg: TCfg) -> None:
             num_workers=cfg["num_workers"],
         )
 
-    loaders_val = DataLoader(dataset=valid_dataset, batch_size=cfg["bs_val"], num_workers=cfg["num_workers"])
+    loader_val = DataLoader(dataset=valid_dataset, batch_size=cfg["bs_val"], num_workers=cfg["num_workers"])
+
+    return loader_train, loader_val
+
 
-    module_kwargs = scheduler_kwargs
+def pl_train(cfg: TCfg) -> None:
+    """
+    This is an entrypoint for the model training in metric learning setup.
+
+    The config can be specified as a dictionary or with hydra: https://hydra.cc/.
+    For more details look at ``examples/README.md``
+
+    """
+    # Here we try to load NEPTUNE_API_TOKEN from .env file
+    # You can also set it up via `export NEPTUNE_API_TOKEN=...`
+    load_dotenv()
+
+    set_global_seed(cfg["seed"])
+
+    cfg = dictconfig_to_dict(cfg)
+    logger = initialize_logging(cfg)
+
+    trainer_engine_params = parse_engine_params_from_config(cfg)
+    is_ddp = check_is_config_for_ddp(trainer_engine_params)
+
+    loader_train, loaders_val = get_retrieval_loaders(cfg)
+    extractor = get_extractor_by_cfg(cfg["model"])
+    criterion = get_criterion_by_cfg(cfg["criterion"], **{"label2category": loader_train.dataset.get_label2category()})
+    optimizable_parameters = [
+        {"lr": cfg["optimizer"]["args"]["lr"], "params": extractor.parameters()},
+        {"lr": cfg["optimizer"]["args"]["lr"], "params": criterion.parameters()},
+    ]
+    optimizer = get_optimizer_by_cfg(cfg["optimizer"], **{"params": optimizable_parameters})  # type: ignore
+
+    module_kwargs = parse_scheduler_from_config(cfg, optimizer=optimizer)
```

Comment:
`module_kwargs = parse_scheduler_from_config` looks strange in terms of naming. But probably we don't want to create an empty dict and fill it